contracts/src/OffChainMessageContracts/ValidatorSetSig.sol

```diff
@@ -0,0 +1,113 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import {
+    WarpMessage,
+    IWarpMessenger
+} from "@avalabs/subnet-evm-contracts@1.2.0/contracts/interfaces/IWarpMessenger.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts@4.8.1/security/ReentrancyGuard.sol";
+
+struct ValidatorSetSigMessage {
+    address validatorSetSigAddress;
+    address targetContractAddress;
+    bytes32 targetBlockChainID;
+    uint256 nonce;
+    bytes txPayload;
+}
+
+/**
+ * @dev Contract that verifies that a set threshold of validators from a given blockchainID
```

Comment:
Let's add a comment that the threshold is set by the validators themselves in their Warp configs: https://github.com/ava-labs/subnet-evm/blob/master/precompile/contracts/warp/config.go#L50